pkgs/applications/misc/udiskie/default.nix

```diff
@@ -0,0 +1,47 @@
+{ stdenv, fetchFromGitHub, asciidoc-full, gettext
+, gobjectIntrospection, gtk3, hicolor_icon_theme, libnotify
+, pythonPackages, udisks2, wrapGAppsHook }:
+
+pythonPackages.buildPythonApplication rec {
+  name = "udiskie-${version}";
+  version = "1.5.1";
+
+  src = fetchFromGitHub {
+    owner = "coldfix";
+    repo = "udiskie";
+    rev = version;
+    sha256 = "01x5fvllb262x6r3547l23z7p6hr7ddz034bkhmj2cqmf83sxwxd";
+  };
+
+  preConfigure = ''
+    export XDG_RUNTIME_DIR=/tmp
```

Comment:
On 12 September 2016 at 11:07, Frederik Rietdijk
<notifications@github.com> wrote:
>
> In pkgs/applications/misc/udiskie/default.nix:
>
> > +, gobjectIntrospection, gtk3, hicolor_icon_theme, libnotify
> > +, pythonPackages, udisks2, wrapGAppsHook }:
> > +
> > +pythonPackages.buildPythonApplication rec {
> > +  name = "udiskie-${version}";
> > +  version = "1.5.1";
> > +
> > +  src = fetchFromGitHub {
> > +    owner = "coldfix";
> > +    repo = "udiskie";
> > +    rev = version;
> > +    sha256 = "01x5fvllb262x6r3547l23z7p6hr7ddz034bkhmj2cqmf83sxwxd";
> > +  };
> > +
> > +  preConfigure = ''
> > +    export XDG_RUNTIME_DIR=/tmp
>
> @bjornfor could you explain why you think it is weird or wrong? I can imagine it is strange to set something outside of a phase, and so it might not be clear when exactly these env vars are set.

Why I think it's weird:
* No clear distinction between Nix value and Shell value. It's just
"all the same".
* A very "global" scope. No way to set an envvar for _just_ configure phase.

I think something like this would be cleaner (inspired by Buildroot):

    builderEnv = { MYVAR = "VALUE"; };    # this will be set for all
processes spawned in a build
    configureEnv = { MYVAR = "VALUE"; };   # just when invoking ./configure

I guess it all comes down to ones perception of the
stdenv.mkDerivation function. Mine is probably wrong :-)